ambry-server/src/main/java/com/github/ambry/server/RepairRequestsSender.java

```diff
@@ -0,0 +1,237 @@
+          requestInfos.clear();
+
+          for (ResponseInfo info : responses) {
+            inflightReqCount--;
+            ReplicateBlobResponse res = (ReplicateBlobResponse) info.getResponse();
+            ReplicateBlobRequest req = (ReplicateBlobRequest) info.getRequestInfo().getRequest();
+            if (res.getError() == ServerErrorCode.No_Error) {
+              RepairRequestRecord.OperationType type;
+              if (req.getOperationType() == RequestOrResponseType.TtlUpdateRequest) {
+                type = RepairRequestRecord.OperationType.TtlUpdateRequest;
+              } else {
+                type = RepairRequestRecord.OperationType.DeleteRequest;
+              }
+              db.removeRepairRequests(req.getBlobId().toString(), type);
+              logger.info("RepairRequests Sender: Repaired {}", req);
+            } else {
+              // LOCAL_CONSISTENCY_TODO: metrics for all the errors happens in RepairRequestsSender.
+              // Ideally we may need create alert if we cannot repair the requests in timely matter.
+              logger.info("RepairRequests Sender: failed to repair {} response {} ", req, res);
+            }
+          }
+
+          if (inflightReqCount > 0) {
+            logger.error("RepairRequests Sender: timeout waiting for repairs {} {}", nodeId, inflightReqCount);
+          }
+        } // loop all the partitions
+      } catch (Throwable e) {
+        // TODO add metric to track background threads
+        logger.error("RepairRequests Sender: Exception when handling request", e);
+        // this is bad and we need to shutdown the app
+        Runtime.getRuntime().halt(1);
+      } finally {
+        if (requestInfos != null) {
+          for (RequestInfo reqInfo : requestInfos) {
+            reqInfo.getRequest().release();
+          }
+          requestInfos = null;
+        }
+      }
+    }
+  }
+
+  /**
+   * Get the {@link RepairRequestRecord} from the database and generate {@link ReplicateBlobRequest}
+   * @param partitionId the partition id
+   * @return list of {@link RequestInfo}
+   */
+  private List<RequestInfo> getAmbryRequest(long partitionId) throws Exception {
+    List<RequestInfo> requestInfos = new ArrayList<>();
+    // get the repair requests for this partition and this node is not the source replica.
+    List<RepairRequestRecord> records = db.getRepairRequests(partitionId, nodeId.getHostname(), nodeId.getPort());
+
+    for (RepairRequestRecord record : records) {
+      BlobId blobId;
+      try {
+        blobId = new BlobId(record.getBlobId(), clusterMap);
+      } catch (IOException e) {
+        // new BlobId may generate IOException
+        logger.error("RepairRequests Sender: Failed to generate Blob ID {}", record, e);
+        continue;
+      }
+
+      ReplicaId replicaId = partition2Replicas.get(partitionId);
+      if (replicaId == null) {
+        logger.error("RepairRequests Sender: Replica is null {} {} {}", partitionId, record, replicaId);
```

Comment:
same above.

Reply:
Will create another PR for all the metrics.